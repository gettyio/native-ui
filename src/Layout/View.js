import styled, { css } from 'styled-components/native'

const margin = css`
  ${props => props.m && css`margin: ${props.theme.spacing[props.m]}px;`}
  ${props => props.mh && css`margin-horizontal: ${props.theme.spacing[props.mh]}px;`}
  ${props => props.mv && css` margin-vertical: ${props.theme.spacing[props.mv]}px;`}
  ${props => props.mt && css` margin-top: ${props.theme.spacing[props.mt]}px;`}
  ${props => props.mb && css` margin-bottom: ${props.theme.spacing[props.mb]}px;`}
  ${props => props.ml && css` margin-left: ${props.theme.spacing[props.ml]}px;`}
  ${props => props.mr && css` margin-right: ${props.theme.spacing[props.mr]}px;`};
`
const padding = css`
  ${props => props.p && css`padding: ${props.theme.spacing[props.p]}px;`}
  ${props => props.ph && css`padding-horizontal: ${props.theme.spacing[props.ph]}px;`}
  ${props => props.pv && css` padding-vertical: ${props.theme.spacing[props.pv]}px;`}
  ${props => props.pt && css` padding-top: ${props.theme.spacing[props.pt]}px;`}
  ${props => props.pb && css` padding-bottom: ${props.theme.spacing[props.pb]}px;`}
  ${props => props.pl && css` padding-left: ${props.theme.spacing[props.pl]}px;`}
  ${props => props.pr && css` padding-right: ${props.theme.spacing[props.pr]}px;`};
`

const View = styled.View`
  ${props => props.flex && css`flex: ${props.flex};`}
  ${props => props.align && css`align-items: ${props.align};`}
  ${props => props.justify && css`justify-content: ${props.justify};`}
  ${props => props.background && css`background: ${props.background};`}
  ${props => props.height && css`height: ${props.height}px;`}
  ${props => props.width && css`width: ${props.width}px;`}
  ${margin}
  ${padding}
`

View.defaultProps = {
  theme: {
    spacing: [2, 4, 8, 16, 32, 64]
  }
}

export default View
