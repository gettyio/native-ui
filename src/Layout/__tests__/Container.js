/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Container from '../Container'
import View from '../View'

describe('Container component', () => {
  it('should render without issue', () => {
    const component = shallow(<Container><View /></Container>)

    expect(component.length).toBe(1)
    expect(toJson(component)).toMatchSnapshot()
  })

  it('should have default props', () => {
    const component = shallow(<Container><View /></Container>)

    expect(toJson(component)).toMatchSnapshot()
    expect(toJson(component)).toHaveStyleRule('flex-grow', 1)
    expect(toJson(component)).toHaveStyleRule('flex-shrink', 1)
  })
})
