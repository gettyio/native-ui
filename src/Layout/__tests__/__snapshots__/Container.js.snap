// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component should have default props 1`] = `
<Component
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
      undefined,
    ]
  }
>
  <Styled(Component) />
</Component>
`;

exports[`Container component should render without issue 1`] = `
<Component
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
      undefined,
    ]
  }
>
  <Styled(Component) />
</Component>
`;
